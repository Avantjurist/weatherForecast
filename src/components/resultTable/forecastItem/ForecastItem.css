.city-data {
    display: flex;
    margin: 10px 0;
}
.temp {
    background-color: #000;
    color: #fff;
}

.city-name {
    background-color: #E8D024;
    color: #fff;
    min-width: 140px;
    text-align: center;
}