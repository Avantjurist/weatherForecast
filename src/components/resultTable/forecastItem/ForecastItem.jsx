import React from 'react';
import * as styles from './ForecastItem.css'

const fromKelToCel = (far) => {
    return Math.round(far - 273.15)
};

const ForecastItem = (props) => {
    return (
        <li className={styles.cityData}>
            <div className={styles.temp}>
                {fromKelToCel(props.item.main.temp) + 'Â°C'}
            </div>
            <div className={styles.cityName}>
                {props.item.name}
            </div>
        </li>
        )
}

export default ForecastItem