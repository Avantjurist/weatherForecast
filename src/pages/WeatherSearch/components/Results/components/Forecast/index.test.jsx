import React from "react";
import { shallow } from "enzyme";
import ForecastItem from "./index.component";

describe("Forecast item", () => {
    it("render corectly with props", () => {
        const props = {
            item: {
                main: {
                    temp: 123
                },
                name: "test"
            }
        };
        const output = shallow(
            <ForecastItem {...props} />
        );

        expect(output).toMatchSnapshot();
    })
})