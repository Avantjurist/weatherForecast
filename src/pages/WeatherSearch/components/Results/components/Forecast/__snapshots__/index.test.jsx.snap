// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast item render corectly with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForecastItem
    item={
      Object {
        "main": Object {
          "temp": 123,
        },
        "name": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
        >
          -150°C
        </div>,
        <div
          className={undefined}
        >
          test
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-150°C",
          "className": undefined,
        },
        "ref": null,
        "rendered": "-150°C",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": undefined,
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          >
            -150°C
          </div>,
          <div
            className={undefined}
          >
            test
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-150°C",
            "className": undefined,
          },
          "ref": null,
          "rendered": "-150°C",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": undefined,
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
